plugin/restore/mysql/mysql.go

```diff
@@ -144,6 +152,156 @@ func getPITROldDatabaseName(database string, suffixTs int64) string {
 	return getSafeName(database, suffix)
 }
 
+// SyncArchivedBinlogFiles syncs the binlogs between the instance and `saveDir`,
+// but exclude latest binlog. We will download the latest binlog only when doing PITR.
+func (r *Restore) SyncArchivedBinlogFiles(ctx context.Context, instance *api.Instance, saveDir string) error {
+	binlogFilesLocal, err := ioutil.ReadDir(saveDir)
+	if err != nil {
+		return err
+	}
+
+	binlogFilesOnServer, err := r.getBinlogFilesMetaOnServer(ctx)
+	if err != nil {
+		return err
+	}
+
+	latestBinlogFileOnServer, err := r.getLatestBinlogFileMeta(ctx)
+	if err != nil {
+		return err
+	}
+
+	// compare file sizes and names to decide which files to download
+	// downloadedName maintains the name of BinlogFile slice, for example:
+	// `SHOW MASTER LOGS` return ["binlog.000001", "binlog.000002"],
+	// if we had downloaded binlog.000001 and file size match, {"binlog.000001":struct{}{}} will be involved in downloadedName.
+	downloadedName := make(map[string]struct{})
+	for _, serverFile := range binlogFilesOnServer {
+		// We don't download the latest binlog in SyncArchivedBinlogFiles()
```

Comment:
The logic seems to be a bit complex 1) skip latest, 2) skip downloaded file and it's size. Instead of calculating downloaded files and skip them again, can we calculate the ones to be downloaded?

sudo code:
```
// build a local file size map from file name to size.
var localFileMap map[string]int64
...
var todo []string
for _, binlog := binlogFilesOnServer {
  if binlog.name == latest.Name {
    continue
  }
  localBinlog, ok := localFileMap[binlog.name]
  if ok {
    if binlog.Size == localBinlog.Size {
      continue
    } else {
      // delete, or maybe we don't need to do it if already handled by mysqlbinlog.
    }
  }
  todo = append(todo, binlog)
}
...
```